spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsFunctionProcessor.java

```diff
@@ -184,6 +189,8 @@ public void orchestrateFunctionInvoking(ResolvableType resolvableType, String fu
 					i++;
 				}
 				if (result != null) {
+					kafkaStreamsBindingInformationCatalogue.addOutputKStreamResolvable(
+							outboundResolvableType != null ? outboundResolvableType : resolvableType.getGeneric(1));
```

Comment:
I am assuming `resolvableType.getGeneric(1)` can't be null or setting  `addOutputKStreamResolvable` to null is ok

Reply:
ya, that's right, in most cases, the `outboundResolvableType` is already set by the pre-processors. Otherwise, we get that from the functions' return signature.